tests/generation/test_generation_tf_logits_process.py

```diff
@@ -114,17 +114,32 @@ def test_repetition_penalty_dist_process(self):
         scores = tf.where(mask, -1 / vocab_size, scores)
         mask = tf.cast(tf.constant([10 * [0], 5 * [0] + [1] + 4 * [0]]), tf.bool)
         scores = tf.where(mask, 4 / vocab_size, scores)
+        return vocab_size, cur_len, input_ids, scores
 
-        rep_penalty_proc = TFRepetitionPenaltyLogitsProcessor(penalty=2.0)
-
-        scores = rep_penalty_proc(input_ids, tf.identity(scores), cur_len)
-
-        # check that values were correctly changed
+    def _check_repetition_penalty_outputs(self, scores, vocab_size):
+        # check that values were correctly changed (negative scores for used tokens should increase, others
+        # should decrease)
         self.assertAlmostEqual(scores[0, 0].numpy(), -(1 / vocab_size) * 2)
         self.assertAlmostEqual(scores[0, 1].numpy(), (1 / vocab_size) / 2)
+        self.assertAlmostEqual(scores[0, 2].numpy(), (1 / vocab_size))  # unused tokens should see no change
 
         self.assertAlmostEqual(scores[1, 0].numpy(), (1 / vocab_size) / 2)
         self.assertAlmostEqual(scores[1, 5].numpy(), (4 / vocab_size) / 2)
+        self.assertAlmostEqual(scores[0, 2].numpy(), (1 / vocab_size))  # unused tokens should see no change
+
+    def test_repetition_penalty_dist_process(self):
+        vocab_size, cur_len, input_ids, scores = self._get_repetition_penalty_inputs()
+        rep_penalty_proc = TFRepetitionPenaltyLogitsProcessor(penalty=2.0)
+        scores = rep_penalty_proc(input_ids, tf.identity(scores), cur_len)
+        self._check_repetition_penalty_outputs(scores, vocab_size)
+
+    @unittest.skipIf(not get_gpu_count(), "XLA not reliable on CPU")
```

Comment:
This should really be solved IMO. Ok to have it like this for now, but the problem is that it's never run on Circle CI so we would only find it when merged to master or for the daily tests.

Reply:
Could we maybe add a big `# TODO` here to clean it up

Reply:
Agreed, it is highly undesirable. Adding the TODO on all related `skipIf`.

Reply:
actually -- removing the skips here, as the issue seems to be exclusively around the softmax (but adding the TODO to the models)